### ISSUE TITLE
Add option to squash UserWarning about double derivatives

### ISSUE THREAD

**Issue by sirmarcel** [lab-cosmo/sphericart#147]
Hello, it seems that #141 added a warning on initialisation about issues with second derivatives. Is it possible to disable this warning, or at least only issue it when the second derivatives are called? It's not a huge issue, but I'm not a big fan of unnecessary entries in my log files.

**Comment by ceriottm** [comment]
This should be triggered only once, and it's actually kind of important as it can lead to silent failures down the line. You can disable it with the usual mechanism you mute warnings, but was discussed at length between @Luthaf and @frostedoyster and I thing this was the least evil. Don't ask me why it can't be triggered when second derivatives are computed, but I think it's a torch limitation (?).

**Comment by sirmarcel** [comment]
Silent failure is indeed bad, is there any reason to not just return `nan` in this case? Seems like silently returning something that is a number is a bit of a foot gun, as they say.

**Comment by frostedoyster** [comment]
Perhaps we could make the switch to `torch.compile`, which should solve these issues. IDK if that would integrate well with our software stack @Luthaf

**Comment by frostedoyster** [comment]
I agree that, as an easier fix, we can have a `warn` argument (true by default)

**Comment by sirmarcel** [comment]
I don't think it's worth adding an argument tbh, I'll just live with it.

**Comment by sirmarcel** [comment]
Alright, I think this is a clear wontfix, I'm closing this.

**Comment by ceriottm** [comment]
yeah one can use `warnings.filterwarnings` if it gets annoying
